### PR TITLE
Updates to the GDP transformations

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This fixes a lingering test failure in Pyomo 5.4.3.  It also adds some improved functionality (see below)

## Changes proposed in this PR:
- Fix a lingering GDP test failure
- Allow implicit declaration of disjunctions containing more than a single constraint
- Switch to using ConfigBlocks for configuring the CHull relaxations.
- Improved test coverage

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/gdp/plugins/cuttingplane.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 )
 from pyomo.opt import SolverFactory
 
-from pyomo.gdp import Disjunct, Disjunction
+from pyomo.gdp import Disjunct, Disjunction, GDP_Error
```
Comment (blnicho) [inline pyomo/gdp/plugins/cuttingplane.py]: You're importing `GDP_Error` but never using it?

File: pyomo/gdp/plugins/gdp_var_mover.py [hunk 2]
```diff
@@ -73,15 +73,21 @@ def _apply_to(self, instance, **kwds):
...
-                for con in cons_in_disjunct:
-                    con.deactivate()
-
             # Reclassify this disjunct as a block
             disjunct_component.parent_block().reclassify_component_type(
                 disjunct_component, Block)
             disjunct_component._activate_without_unfixing_indicator()
+
+            # Deactivate all constraints.  Note that we only need to
+            # descent into blocks: we will catch disjuncts in the outer
```
Comment (blnicho) [inline pyomo/gdp/plugins/gdp_var_mover.py]: typo "need to descent into"

File: pyomo/gdp/tests/models.py [hunk 3]
```diff
@@ -162,29 +162,34 @@ def makeTwoTermDisjOnBlock():
     m = ConcreteModel()
     m.b = Block()
     m.a = Var(bounds=(0,5))
-    def d_rule(disjunct, flag):
+
+    # On a whim, verify that the decorator notation works
+    @m.b.Disjunct([0,1])
```
Comment (blnicho) [inline pyomo/gdp/tests/models.py]: Where is the decorator defined? Is this relying on the same decorator as `Constraint`?

File: pyomo/gdp/tests/test_disjunct.py [hunk 4]
```diff
@@ -48,26 +48,62 @@ def test_construct_implicit_disjuncts(self):
...
+        self.assertIs(disjuncts[0].constraint[1].body, m.y)
         self.assertIs(disjuncts[1].parent_block(), m)
-        self.assertIs(disjuncts[1].constraint.body, m.x)
-        self.assertEqual(len(disjuncts[0].parent_component().name), 4)
-        self.assertEqual(disjuncts[0].name[:2], "e_")
+        self.assertIs(disjuncts[1].constraint[1].body, m.x)
+        self.assertEqual(len(disjuncts[0].parent_component().name), 13)
+        self.assertEqual(disjuncts[0].name[:12], "e_disjuncts_")
+
+        # Test that the implicit disjuncts acn be lists/tuples
```
Comment (blnicho) [inline pyomo/gdp/tests/test_disjunct.py]: typo "acn"

File: pyomo/gdp/util.py [hunk 5]
```diff
@@ -72,3 +75,27 @@ def clone_without_expression_components(expr, substitute):
                 _idx = 0
                 _len = len(_argList)
     return ans[0]
+
+
+def target_list(x):
```
Comment (blnicho) [inline pyomo/gdp/util.py]: This seems like a useful function even outside of GDP. Maybe move this to general Pyomo utils?

File: pyomo/gdp/plugins/cuttingplane.py (hunk 1 above)
Comment (jsiirola) [reply pyomo/gdp/plugins/cuttingplane.py]: GDP_Error is raised below.  This is fixing a bug/test failure from the 5.4.3 release.

File: pyomo/gdp/tests/models.py (hunk 3 above)
Comment (jsiirola) [reply pyomo/gdp/tests/models.py]: Yes - this makes use of the standard decorator functionality.  The `@model.Constraint(...)` notation works for any component that takes a rule as a keyword argument.

File: pyomo/gdp/util.py (hunk 5 above)
Comment (jsiirola) [reply pyomo/gdp/util.py]: Maybe after a little more refinement.  Working through this function highlighted some "challenges" with transformations that I'd like to think through before promulgating this as the recommended solution.

**Pull request merged**